### DISCUSSION TITLE
gatsby-plugin-sitemap does not generate sitemap

### DISCUSSION THREAD

**Discussion by bit-crafter** [gatsbyjs/gatsby#33904] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### configuration
```
      {
            resolve: `gatsby-plugin-sitemap`,
            options: {
                excludes: [___PAGES___],
            },
        },
```

### Reproduction Link

https://github.com/tumdav

### Steps to Reproduce

1. Install gatsby 4.1.0
2. Install the latest version of gatsby-plugin-sitemap
3. yarn build & yarn serve
4. Open a browser window and check localhost:9000/sitemap.xml or just try to find a file inside of build folder

### Expected Result

On production build, it should generate a sitemap.xml file

### Actual Result

Sitemap.xml file is missing  after production build

### Environment

```shell
System:
    OS: macOS 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.15.0 - ~/.nvm/versions/node/v14.15.0/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v14.15.0/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 95.0.4638.69
    Firefox: 87.0
    Safari: 14.1.2
  npmPackages:
    gatsby: ^4.1.0 => 4.1.0
    gatsby-background-image: ^1.5.3 => 1.5.3
    gatsby-plugin-anchor-links: ^1.2.1 => 1.2.1
    gatsby-plugin-canonical-urls: ^4.1.0 => 4.1.0
    gatsby-plugin-catch-links: ^4.1.0 => 4.1.0
    gatsby-plugin-google-tagmanager: ^4.1.0 => 4.1.0
    gatsby-plugin-image: ^2.1.0 => 2.1.0
    gatsby-plugin-manifest: ^4.1.0 => 4.1.0
    gatsby-plugin-meta-redirect: ^1.1.1 => 1.1.1
    gatsby-plugin-no-sourcemaps: ^4.1.0 => 4.1.0
    gatsby-plugin-nprogress: ^4.1.0 => 4.1.0
    gatsby-plugin-preact: ^6.1.0 => 6.1.0
    gatsby-plugin-react-helmet: ^5.1.0 => 5.1.0
    gatsby-plugin-robots-txt: ^1.6.14 => 1.6.14
    gatsby-plugin-sharp: ^4.1.0 => 4.1.0
    gatsby-plugin-sitemap: 5.2.0-next.1 => 5.2.0-next.1
    gatsby-source-filesystem: ^4.1.0 => 4.1.0
    gatsby-transformer-json: ^4.1.0 => 4.1.0
    gatsby-transformer-sharp: ^4.1.0 => 4.1.0
```

**Comment by LekoArts** [comment, marked as answer]
Hi!

Please see: https://www.gatsbyjs.com/plugins/gatsby-plugin-sitemap/

You should look at `/sitemap/sitemap-index.xml` and `/sitemap/sitemap-0.xml`

**Reply by PranjalEPYC** [reply]
Thanks man this worked

**Reply by khushi-webforest** [reply]
Hey @LekoArts 

> I'm getting this two files in **local** after yarn build & yarn serve :

`/sitemap/sitemap-index.xml` and `/sitemap/sitemap-0.xml`

> And  in **production** I'm getting `https://www.example.com/sitemap/sitemap-index.xml`

> according to document it should be `https://www.example.com/sitemap-index.xml`

can you please help me ? 
what step I'm missing here ?

**Reply by Histamina** [reply]
Browsing I've found that if you downgrade the `gatsby-plugin-sitemap` to `"3.3.0"`, works and generates the `sitemap.xml` file.
Have in mind that my dependencies versions are these:

`
"gatsby": "5.1.0",
"gatsby-plugin-sitemap": "3.3.0",
"gatsby-plugin-robots-txt": "^1.7.1",
`

And still works and I don't have compatibility issues. Hope this helps.
